### ISSUE TITLE
Using terraform from an EC2 instance with EC2 provisioning role

### ISSUE THREAD

**Issue by jitingupta007** [terraform-providers/terraform-provider-aws#7935]
Hi , 

I want to provision EC2 services using Terraform but without using the Secret and Access keys instead using IAM role.

Can anyone help me with the related Terraform configuration, How it can do provisioning from an EC2 instance (with required IAM role) ??

**Comment by dannyleesmith** [comment]
Hi @jitingupta007, I found this that looks related:

> Note: If you simply leave out AWS credentials, Terraform will automatically search for saved API credentials (for example, in ~/.aws/credentials) or IAM instance profile credentials. This option is much cleaner for situations where tf files are checked into source control or where there is more than one admin user. See details here. Leaving IAM credentials out of the Terraform configs allows you to leave those credentials out of source control, and also use different IAM credentials for each user without having to modify the configuration files.
-- https://learn.hashicorp.com/terraform/getting-started/build

And further detail here: https://www.terraform.io/docs/providers/aws/#ec2-role

TL;DR: in the absence of all other credentials it will query the EC2 instance profile and then use that.

Hope that helps!

-- Danny

**Comment by tracypholmes** [comment]
Thank you for using Terraform and for opening up this question @jitingupta007! It appears your question has been answered. As a reminder, issues on GitHub are intended to be related to bugs or feature requests with the provider codebase. Please use https://discuss.hashicorp.com/c/terraform-providers for community discussions, and questions around Terraform.

If you believe that your issue was miscategorized as a question or closed in error, please create a new issue using one of the following provided templates: [bug report](https://github.com/terraform-providers/terraform-provider-aws/issues/new?template=Bug_Report.md) or [feature request](https://github.com/terraform-providers/terraform-provider-aws/issues/new?labels=enhancement&template=Feature_Request.md). Please make sure to provide us with the appropriate information so we can best determine how to assist with the given issue.
